File: frontend/src/types/connection.ts
/**
 * TypeScript types for database connections.
 */

export interface Connection {
  id: number;
  name: string;
  host: string;
  port: number;
  database: string;
  username: string;
  description?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ConnectionCreate {
  name: string;
  host: string;
  port: number;
  database: string;
  username: string;
  password: string;
  description?: string;
}

export interface ConnectionUpdate {
  name?: string;
  host?: string;
  port?: number;
  database?: string;
  username?: string;
  password?: string;
  description?: string;
}

export interface ConnectionTestRequest {
  connection_id?: number;
  host?: string;
  port?: number;
  database?: string;
  username?: string;
  password?: string;
}

export interface ConnectionTestResponse {
  success: boolean;
  message: string;
  error?: string;
}

File: backend/src/models/connection.py
"""SQLAlchemy model for database connections."""
from datetime import datetime

from sqlalchemy import Integer, String, Text, DateTime
from sqlalchemy.orm import Mapped, mapped_column

from src.utils.database import Base


class Connection(Base):
    """Database connection model."""
    
    __tablename__ = "connections"
    
    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(255), unique=True, nullable=False, index=True)
    host: Mapped[str] = mapped_column(String(255), nullable=False)
    port: Mapped[int] = mapped_column(Integer, nullable=False)
    database: Mapped[str] = mapped_column(String(255), nullable=False)
    username: Mapped[str] = mapped_column(String(255), nullable=False)
    password_encrypted: Mapped[str] = mapped_column(Text, nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=datetime.utcnow, onupdate=datetime.utcnow
    )
    
    def __repr__(self) -> str:
        """String representation of Connection."""
        return f"<Connection(id={self.id}, name='{self.name}', host='{self.host}')>"
